From 7e6d465fbd227f65ccf05e8b76541b9c95cc405f Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Tue, 10 Oct 2017 16:59:50 +0800
Subject: [PATCH 2616/5242] MLK-16563-1: ASoC: cs42xx8: reset the codec in the
 beginning of probe

commit  3041c698bb7763010e00a4be92a65f008a6a7f31 from
https://source.codeaurora.org/external/imx/linux-imx.git

In AUDIO IO board for imx8qxp mek, after board reset, the codec
failed to probe, system can't find codec device on i2c bus.
The reason is not clear, but add reset operation in the beginning of
probe can fix this issue.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/cs42xx8.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/cs42xx8.c b/sound/soc/codecs/cs42xx8.c
index 7b49e14..49e8646 100644
--- a/sound/soc/codecs/cs42xx8.c
+++ b/sound/soc/codecs/cs42xx8.c
@@ -535,11 +535,12 @@ int cs42xx8_probe(struct device *dev, struct regmap *regmap)
 	cs42xx8->reset_gpio = of_get_named_gpio(np, "reset-gpio", 0);
 	if (gpio_is_valid(cs42xx8->reset_gpio)) {
 		ret = devm_gpio_request_one(dev, cs42xx8->reset_gpio,
-				GPIOF_OUT_INIT_HIGH, "cs42xx8 reset");
+				GPIOF_OUT_INIT_LOW, "cs42xx8 reset");
 		if (ret) {
 			dev_err(dev, "unable to get reset gpio\n");
 			return ret;
 		}
+		gpio_set_value_cansleep(cs42xx8->reset_gpio, 1);
 	}
 
 	cs42xx8->clk = devm_clk_get(dev, "mclk");
@@ -622,8 +623,10 @@ static int cs42xx8_runtime_resume(struct device *dev)
 		return ret;
 	}
 
-	if (gpio_is_valid(cs42xx8->reset_gpio))
+	if (gpio_is_valid(cs42xx8->reset_gpio)) {
+		gpio_set_value_cansleep(cs42xx8->reset_gpio, 0);
 		gpio_set_value_cansleep(cs42xx8->reset_gpio, 1);
+	}
 
 	ret = regulator_bulk_enable(ARRAY_SIZE(cs42xx8->supplies),
 				    cs42xx8->supplies);
-- 
1.7.9.5

